components/resources/library/src/skikoMain/kotlin/org/jetbrains/compose/resources/FontResources.skiko.kt

```diff
@@ -29,12 +31,27 @@ private val emptyFontBase64 =
 @OptIn(ExperimentalEncodingApi::class)
 private val defaultEmptyFont by lazy { Font("org.jetbrains.compose.emptyFont", Base64.decode(emptyFontBase64)) }
 
+private val cacheMutex = Mutex()
+private val fontsCachedBySkiko = mutableSetOf<String>()
+
 @Composable
 actual fun Font(resource: FontResource, weight: FontWeight, style: FontStyle): Font {
     val resourceReader = LocalResourceReader.current
     val fontFile by rememberResourceState(resource, weight, style, { defaultEmptyFont }) { env ->
         val path = resource.getResourceItemByEnvironment(env).path
-        val fontBytes = resourceReader.read(path)
+
+        // Skiko has own internal font's cache,
+        // so we can provide an empty byte array since the font was cached already
+        val fontBytes = cacheMutex.withLock {
+            if (fontsCachedBySkiko.contains(path)) {
+                ByteArray(0)
```

Comment:
I looked at the sources, and `getData` is called by Compose (not Skia) cache [here](https://github.com/JetBrains/compose-multiplatform-core/blob/f9b90c58dfb166be398917f1e43326f6566e50ff/compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/platform/PlatformFont.skiko.kt#L211). It uses `ExpireAfterAccessCache` with 1-minute expiration date.

So, we can't guarantee that this data is stored forever.